vimms/Controller/dia.py

```diff
@@ -33,9 +33,12 @@ def __init__(self,min_mz,max_mz,
 
         self.scan_number = self.initial_scan_id
 
+        self.make_all_scans_ms1= make_all_scans_ms1 #flag to control if ms2 to ms1 for all ion fragmentation
+
     # method required by super-class
     def update_state_after_scan(self, last_scan):
-        pass
+        if self.make_all_scans_ms1 == True and last_scan.ms_level == 2: #do ms2 to ms1 hack if flag on
```

Comment:
do we still need this if the correction is in the MS class?

Reply:
can remove checking as MS hack code can serve same purpose, code updated to rollback hack code. will test & commit again.